napture/src/main.rs

```diff
@@ -374,10 +411,14 @@ fn fetch_dns(url: String) -> String {
         let status = response.status();
 
         if let Ok(json) = response.json::<DomainInfo>() {
-            json.ip
+            return json.ip
         } else {
             lualog!("debug", format!("Failed to parse response body from DNS API. Error code: {}. Returning original URL.", status.as_u16()));
-            url
+            if status.as_u16() == 404 {
+                URI_PARAMETERS.lock().unwrap().insert("q".to_string(), url.clone());
+                return "https://github.com/face-hh/dingle-frontend".to_string();
+            }
```

Comment:
What about other http error statuses ?

Reply:
The reason I didn't handle any other http statuses is because if it exists and something is wrong with the server we don't want to act like it doesn't exist.

Reply:
Ah okay but probably should bring up some sort of error page

Reply:
Good point.